docs/source/developer/architecture.rst

```diff
@@ -431,14 +431,177 @@ ufunc-like function that is inlined into a single loop that only
 allocates a single result array.
 
 
+.. _`parallel-accelerator`:
+
+Stage 6b: Perform Automatic Parallelization
+-------------------------------------------
+
+This pass is only performed if the parallel option in the :func:`~numba.jit`
+decorator is set to True.  This pass find parallelism implicit in the 
+semantics of operations in the Numba IR and replaces those operations
+with explicitly parallel representations of those operations using a
+special `parfor` operator.  Then, optimizations are performed to maximize
+the number of parfors that are adjacent to each other such that they can
+then be fused together into one parfor that takes only one pass over the
+data and will thus typically have better cache performance.  Finally,
+during lowering, these parfor operators are converted to a form similar
+to guvectorize to implement the actual parallelism.  All these processes
+are described in more detail in the following paragraphs.
+
+The automatic parallelization pass has a number of sub-passes.
+
+1. CFG simplification - Sometimes Numba IR will contain chains of blocks
+containing no loops which are merged in this sub-pass into single blocks.
+This sub-pass simplifies subsequent analysis of the IR.
+
+2. Numpy canonicalization - Some Numpy operations can be written as
+operations on Numpy objects (e.g., arr.sum()), or as calls to Numpy
+taking those objects (e.g., numpy.sum(arr)).  This sub-pass converts all
+such operations to the latter form for cleaner subsequent analysis.
+
+3. Array analysis - A critical requirement for later parfor fusion is that 
+parfors have identical iteration spaces and these iteration spaces
```

Comment:
Thanks, I obviously didn't make the differentiation. Is there a seminal reference for this array analysis method that could perhaps be added?